analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd

```diff
@@ -0,0 +1,211 @@
+    select(-Kids_First_Biospecimen_ID_DNA) %>%
+    filter(!is.na(Kids_First_Biospecimen_ID_RNA)) %>%
+    rename(Kids_First_Biospecimen_ID = Kids_First_Biospecimen_ID_RNA)
+  
+  dna_df <- results_df %>%
+    select(-Kids_First_Biospecimen_ID_RNA) %>%
+    filter(!is.na(Kids_First_Biospecimen_ID_DNA)) %>%
+    rename(Kids_First_Biospecimen_ID = Kids_First_Biospecimen_ID_DNA)
+  
+  # sometimes there will be multiple RNA-seq assays per single DNA-seq assay
+  # or vice versa, which is why this distinct is necessary
+  return(bind_rows(rna_df, dna_df) %>% distinct())
+}
+```
+
+### Directories and files
+
+```{r}
+data_dir <- file.path("..", "..", "data")
+analyses_dir <- ".."
+
+# directories for upstream subtyping modules
+ews_dir <- file.path(analyses_dir, "molecular-subtyping-EWS")
+hgg_dir <- file.path(analyses_dir, "molecular-subtyping-HGG")
+lgat_dir <- file.path(analyses_dir, "molecular-subtyping-LGAT")
+embryonal_dir <- file.path(analyses_dir, "molecular-subtyping-embryonal")
+
+# the folder that contains the tabular results is standardized across modules
+results_dir <- "results"
+# create the results directory for this module if it does not yet exist
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+#### Input files
+
+When we run this locally, we want to tie it to a specific version of the histologies file _prior_ to feedback we received from pathology that is recorded on [#609](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609).
+
+```{r}
+if (running_in_ci) {
+  histologies_file <- file.path(data_dir, "pbta-histologies.tsv")
+} else {
+  histologies_file <- file.path(data_dir, "release-v15-20200228",
+                                "pbta-histologies.tsv")
+}
+```
+
+Results files from individual modules.
+
+```{r}
+ews_results_file <- file.path(ews_dir, results_dir, "EWS_results.tsv")
+hgg_results_file <- file.path(hgg_dir, results_dir, "HGG_molecular_subtype.tsv")
+lgat_results_file <- file.path(lgat_dir, results_dir, "lgat_subtyping.tsv")
+embryonal_results_file <- file.path(embryonal_dir, results_dir,
+                                    "embryonal_tumor_molecular_subtypes.tsv")
+```
+
+#### Output file
+
+```{r}
+output_file <- file.path(results_dir, "compiled_molecular_subtypes.tsv")
+```
+
+## Read in data
+
+```{r message=FALSE}
+histologies_df <- read_tsv(histologies_file, guess_max = 10000)
+ews_results_df <- read_tsv(ews_results_file)
+hgg_results_df <- read_tsv(hgg_results_file)
+lgat_results_df <- read_tsv(lgat_results_file)
+embryonal_results_df <- read_tsv(embryonal_results_file)
+```
+
+## Compile the subtyping resutls
+
+### Handling non-ATRT/non-MB embryonal tumors
+
+The molecular subtyping information from these tumors went into the v15 release, so we can use the `integrated_diagnosis`, `short_histology`, `broad_histology`, and `Notes` columns from the histologies file from that release.
+
+```{r}
+embryonal_results_df <- bind_exp_strategies(embryonal_results_df) %>%
```

Comment:
embryonal_results_df[,"short_histology"] %>% as.data.frame() %>% table()
Embryonal Tumor | ETMR 
--- | ---
39  | 9 

I guess it's not applicable to code in this notebook but since this notebook is standardizing the molecular_subtyping results should we look into the short_histology here it's "Embryonal Tumor" which isn't quite like other "short_histologies"

Reply:
I believe this is probably intentional/as expected because I imagine `ETMR` is only referring to Embryonal tumors with multilayered rosettes the same way we use `ATRT` in `short_histology`, but I can double check.

Reply:
Yes I think this is consistent with what we'd expect
```r
> histologies_df %>% 
    filter(broad_histology == "Embryonal tumor") %>% 
    group_by(integrated_diagnosis, short_histology) %>% 
    tally()
# A tibble: 4 x 3
# Groups:   integrated_diagnosis [4]
  integrated_diagnosis                     short_histology     n
  <chr>                                    <chr>           <int>
1 Atypical Teratoid Rhabdoid Tumor         ATRT               60
2 CNS Embryonal Tumor                      Embryonal Tumor    39
3 Embryonal tumor with multilayer rosettes ETMR                9
4 Medulloblastoma                          Medulloblastoma   226
```

Reply:
ok cool, thanks for checking!